### DISCUSSION TITLE
Accessibility concerns regarding components with only icons as content

### DISCUSSION THREAD

**Discussion by aweell** [dxc-technology/halstack-react#624] [Ideas] [answered]
For some components (toggle group, button, etc...) that can be rendered only with icons, there's no possibility to add an associate label to indicate the context o purpose of the action. This is an accessibility problem and should be fixed.

Since it's impossible to determine the content, is the component user's (developer) responsibility to add a meaningful title to the visual content, in order to provide information about the context/action to the assistive technologies.

![Screen Shot 2022-02-04 at 11 07 08 AM](https://user-images.githubusercontent.com/44420072/152510660-e8b84851-1282-48e6-93af-b602e37c41d3.png)

_Fig. 1 - Toggle group with only icons as content_

The components that can render icons or images without an associated label should be identified.

## Proposed solutions

* Use a `visually-hidden` `<label>` element
* Use `aria-label` 

Both solutions would require a prop where the user can provide the content that would be passed to these elements.

## References

* [Nont-text content - Understanding SC 1.1.1](https://www.w3.org/TR/2016/NOTE-UNDERSTANDING-WCAG20-20161007/text-equiv-all.html)
* [Sara Soueidan - Accesible icon buttons](https://www.sarasoueidan.com/blog/accessible-icon-buttons/)

**Comment by GomezIvann** [comment, marked as answer]
I'm fully agree with this discussion. We need to review only-icon components.

**Reply by GomezIvann** [reply]
An issue (https://github.com/dxc-technology/halstack-react/issues/1490) has been created to explore this!

**Reply by Mil4n0r** [reply]
This has been fixed since the accessibility test PR. We can close this discussion
